cmd/mcp-grafana/main.go

```diff
@@ -135,10 +144,15 @@ func run(transport, addr, basePath, endpointPath string, logLevel slog.Level, dt
 			return fmt.Errorf("server error: %v", err)
 		}
 	case "streamable-http":
-		srv := server.NewStreamableHTTPServer(s, server.WithHTTPContextFunc(mcpgrafana.ComposedHTTPContextFunc(gc)),
+		opts := []server.StreamableHTTPOption{
+			server.WithHTTPContextFunc(mcpgrafana.ComposedHTTPContextFunc(gc)),
 			server.WithStateLess(true),
 			server.WithEndpointPath(endpointPath),
-		)
+		}
+		if tls.certFile != "" || tls.keyFile != "" {
```

Comment:
```suggestion
		if tls.certFile != "" && tls.keyFile != "" {
```

Just a nit, but this seems to be a condition in mcp-go. You can also leave it, to actually return mcp-go's error message.

https://github.com/mark3labs/mcp-go/blob/35fc389d56010b24ef49f966868b8b948d2af48e/server/streamable_http.go#L204-L215